turbopack/packages/devlow-bench/src/interfaces/snowflake.ts

```diff
@@ -0,0 +1,117 @@
+import type { Interface } from "../index.js";
+import os from "os";
+import { CPU_ARCH, CPU_MODEL, GIT_BRANCH, GIT_SHA, IS_CI, NODE_VERSION, NUM_CPUS, OS, OS_RELEASE, USERNAME } from "./constants.js";
+import { randomUUID } from "crypto";
+
+type DevlowMetric = {
+  event_time: number;
+  scenario: string;
+  props: Record<string, string | number | boolean | null>;
+  metric: string;
+  value: number;
+  unit: string;
+  relative_to?: string;
+  is_ci: boolean;
+  os: string;
+  os_release: string;
+  cpus: number;
+  cpu_model: string;
+  user: string;
+  arch: string;
+  total_memory: number;
+  node_version: string;
+  git_sha: string;
+  git_branch: string;
+}
+
+export default function createInterface({
+  gatewayUri = process.env.SNOWFLAKE_BATCH_URI,
+  topicName = process.env.SNOWFLAKE_TOPIC_NAME,
+  schemaId = process.env.SNOWFLAKE_SCHEMA_ID ? parseInt(process.env.SNOWFLAKE_SCHEMA_ID, 10) : undefined,
+}: { gatewayUri?: string; topicName?: string, schemaId?: number } = {}): Interface {
+  if (!gatewayUri)
+    throw new Error("Snowflake gateway URI is required (set SNOWFLAKE_GATEWAY_URI)");
+  if (!topicName)
+    throw new Error("Snowflake topic name is required (set SNOWFLAKE_TOPIC_NAME)");
+  if (!schemaId)
+    throw new Error("Snowflake schema ID is required (set SNOWFLAKE_SCHEMA_ID to a valid integer)");
+
+  const records: DevlowMetric[] = []
+  const iface: Interface = {
+    measurement: async (scenario, props, name, value, unit, relativeTo) => {
+      const ts = Math.round(Date.now() / 1000);
+      records.push({
+        event_time: ts,
+        scenario,
+        props,
+        metric: name,
+        value,
+        unit,
+        relative_to: relativeTo,
+        is_ci: IS_CI,
+        os: OS,
+        os_release: OS_RELEASE,
+        cpus: NUM_CPUS,
+        cpu_model: CPU_MODEL,
+        user: USERNAME,
+        arch: CPU_ARCH,
+        total_memory: Math.round(os.totalmem() / 1024 / 1024 / 1024),
```

Comment:
This is in gigabytes? That doesn't seem like enough precision?

Reply:
This originally comes from the dev implementation. But yeah, no idea why this is gigabytes, aside from maybe wanting to coalesce small differences into the same bucket. Feels like a concern that should be handled when visualizing.